src/status_im/native_module/core.cljs

```diff
@@ -278,6 +279,52 @@
              :connection-string connection-string})
   (.inputConnectionStringForBootstrapping ^js (status) connection-string config-json callback))
 
+(defn deserialize-and-compress-key
+  "Provides a community id public key to status-go and gets back a key that status-mobile understands"
+  [desktop-key callback]
+  (log/info "[native-module] Compressing public key"
+            {:fn          :convert-desktop-key-to-mobile-key
+             :desktop-key desktop-key})
+  (.deserializeAndCompressKey ^js (status) desktop-key callback))
+
+
+(defn serialise-public-key-into-compressed-key
+  "Provides public key to status-go and gets back a compressed key"
+  [public-key callback]
+  (let [serialisation-key constants/serialization-key
+        multi-code-prefix constants/multi-code-prefix
+        multi-code-key    (str multi-code-prefix (subs public-key 2))]
+    (log/info "[native-module] Compressing public key"
+              {:fn             :serialise-public-key-into-compressed-key
+               :public-key     public-key
+               :multi-code-key multi-code-key})
+    (.multiformatSerializePublicKey ^js (status) multi-code-key serialisation-key callback)))
+
+(defn deserialise-compressed-key-into-public-key
+  "Provides compressed key to status-go and gets back the uncompressed public key"
+  [public-key callback]
+  (let [deserialisation-key constants/de-serialization-key]
+    (log/info "[native-module] Deserialising compressed key"
+              {:fn         :deserialise-compressed-key-into-public-key
+               :public-key public-key})
+    (.multiformatDeserializePublicKey ^js (status) public-key deserialisation-key callback)))
+
+(defn decompress-key-into-public-key
+  "Provides compressed key to status-go and gets back the uncompressed public key"
+  [public-key callback]
+  (log/info "[native-module] Decompressing compressed key"
+            {:fn         :deserialise-compressed-key-into-public-key
+             :public-key public-key})
+  (.decompressPublicKey ^js (status) public-key callback))
+
+(defn compress-given-key
```

Comment:
A couple things about this function:

1. Why not just `compress-key`?, The `given` word seems superfluous.
2. The parameter `some-key` uses the word `some`, which is superfluous. Just `key` seems appropriate. `some-`something can be used in almost any parameter anywhere, like `some-callback`, but it's just noise. 
3. The log message strictly says `:public-key`, but the docstring says "Provides ANY key". Also the native call explicitly uses a method named `compressPublickKey`. So is it really *any key* or is it a *public key*? I think we should adjust the code to be clear.